src/twisted/web/http.py

```diff
@@ -1206,29 +1220,43 @@ def write(self, data):
             # persistent connections.
             if (
                 (version == b"HTTP/1.1")
-                and (self.responseHeaders.getRawHeaders(b"content-length") is None)
+                and (
+                    self.responseHeaders._getRawHeadersFaster(
+                        _ENCODED_CONTENT_LENGTH_HEADER
+                    )
+                    is None
+                )
                 and self.method != b"HEAD"
                 and self.code not in NO_BODY_CODES
             ):
-                self.responseHeaders.setRawHeaders("Transfer-Encoding", [b"chunked"])
+                self.responseHeaders._setRawHeadersFaster(
+                    _ENCODED_TRANSFER_ENCODING_HEADER, [b"chunked"]
+                )
```

Comment:
Here we have a static value...so I don't think that it helps  to do the space normalization

```suggestion
                self.responseHeaders.setUnsecure(
                    _ENCODED_TRANSFER_ENCODING_HEADER, [b"chunked"]
                )
```